src/system.h

```diff
@@ -38,7 +42,9 @@ struct Image {
 
     wxBitmap &Display() {
         if (!bm_display.IsOk()) {
-            ScaleBitmap(bm_orig, 1.0 / display_scale * sys->frame->csf, bm_display);
+            wxBitmapType it = imagetypes[image_type].first;
+            wxBitmap bm = ConvertVectorToBitmap(image_data, it);

```

Comment:
So, effectively the cost at loading of converting to bitmap is delayed until here.